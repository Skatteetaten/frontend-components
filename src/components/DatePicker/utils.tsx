export const DEFAULTFORMATDATE = (date: Date | null | undefined): string => {
  return date
    ? Intl.DateTimeFormat('no', {
        year: 'numeric',
        month: '2-digit',
        day: 'numeric',
      }).format(date)
    : '';
};

export const DEFAULTPARSEDATEFROMSTRING = (date: string): Date | null => {
  try {
    if (date && date.length === 10) {
      const day: number = parseInt(date.substring(0, 2));
      const month: number = parseInt(date.substring(3, 5));
      const year: number = parseInt(date.substring(6, 10));
      const formattedDate: Date = new Date(Date.UTC(year, month - 1, day));
      return formattedDate;
    }
  } catch (error) {
    console.log(error);
  }
  return null;
};

export const monthsForLocale = (
  localeName = 'no',
  monthFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { month: monthFormat })
    .format;
  const months: string[] = [];
  for (let month = 0; month < 12; month++) {
    months.push(format(new Date(Date.UTC(2020, month))));
  }
  return months;
};

export const weekdaysForLocale = (
  localeName = 'no',
  dayFormat: 'long' | 'short' = 'long'
) => {
  const format = new Intl.DateTimeFormat(localeName, { weekday: dayFormat })
    .format;
  const weekdays: string[] = [];
  for (let day = 0; day < 7; day++) {
    weekdays.push(format(new Date(Date.UTC(2020, 7, day + 2)))); // add 2 days to let the week start on Sunday
  }
  return weekdays;
};
