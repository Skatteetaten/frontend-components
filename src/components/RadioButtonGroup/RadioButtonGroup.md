** En valggruppe (med radioknapper) brukes i skjemaer for å velge ett alternativ blant flere gjensidig utelukkende valg. **

```js
import { RadioButtonGroup } from '@skatteetaten/frontend-components';

const initialState = {
  options: [
    {
      key: 'A',
      text: 'Enkeltpersonsforetak',
    },
    {
      key: 'B',
      text: 'Aksjeselskap',
    },
  ],
};
<RadioButtonGroup
  label="Type virksomhet"
  options={state.options}
  onChange={(e, option) => console.log(option)}
  help="Type virksomhet vil påvirke hva du må rapportere til oss."
  id="radio"
/>;
```

Med beskrivelse på hvert steg:

```js
import { RadioButtonGroup } from '@skatteetaten/frontend-components';

const initialState = {
  options: [
    {
      key: 'A',
      text: 'Sparekonto',
      description:
        'Dette er egentlig bare en brukskonto, med et annet navn. Du har ubegrenset uttak i året',
    },
    {
      key: 'B',
      text: 'Sparekonto pluss',
      description:
        'Hvis du er veldig glad i å spare uten renter er dette kontoen for deg. Uttak fra denne kontoen kommer med gebyr',
    },
  ],
};
<>
  <RadioButtonGroup
    label="Type sparekonto"
    options={state.options}
    onChange={(e, option) => console.log(option)}
    id="radio"
  />
</>;
```

```js noeditor uu
<h3>Tips</h3>
<ul>
<li>RadioButtonGroup egner seg best når det er 2–7 valg. Hvis det er mer enn 7 valg, bruk heller komponenten Dropdown.</li>
<li>Standard HTML er tilstrekkelig for riktig koding: bruk av fieldset og legend for gruppering av radioknapper under en ledetekst, og input med label for-attributt.</li>
<li>Det skal kun være ett tabstopp på radioknappene med bruk av tastatur. For å bytte mellom valg av radioknapp skal piltast opp/ned brukes. Hvis du tabulerer vekk fra radioknappene og tilbake igjen, skal tastaturfokus være på valgt radioknapp.</li>
<li>Sjekk at tilstand på avhuking (valgt/ikke valgt) leses opp av skjermleser.</li>
</ul>

<h3>Mest relevante WCAG-krav</h3>
<ul>
<li>1.3.1 A, Informasjon og relasjoner </li>
<li>3.3.2 A, Ledetekster eller instruksjoner</li>
<li>4.1.2 A, Navn, rolle, verdi</li>
</ul>

<h3>WAI-ARIA</h3>
<ul>
<li>N/A</li>
</ul>
```

```js noeditor beskrivelse
  <h3>Velge kun ett valg</h3>

  <p>
    Radioknapper bruker vi når ber brukeren om å velge nøyaktig ett valg i en
    liste.
  </p>
  <p>
    Hvis du har plass på siden, og 2-7 valg å velge mellom, er det ofte en
    fordel å bruke slike radioknapper fremfor nedtrekksmeny.
  </p>
  <p>
    Dersom brukeren må ta et aktivt valg, kan listen med radioknappen starte
    uten at noen av valgene er avhuket. Implisitt betyr dette at dersom listen
    med radioknapper får en default-valg satt, MÅ ikke brukeren ta et valg.
  </p>
```
