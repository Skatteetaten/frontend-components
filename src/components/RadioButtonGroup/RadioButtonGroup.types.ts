import { IChoiceGroupOption, IChoiceGroupProps } from '@fluentui/react';
import { calloutState, LabelWithCalloutProps } from '../LabelWithCallout';

export interface IRadioButtonGroupOptions extends IChoiceGroupOption {
  description?: string;
}

export interface RadioButtonGroupProps extends IChoiceGroupProps {
  calloutFloating?: boolean;
  /** CSS class */
  className?: string;
  /** Feilmelding */
  errorMessage?: JSX.Element | string;
  /** Hjelpetekst */
  help?: JSX.Element | string;
  /** Horizontal layout */
  horizontal?: boolean;
  /** Om merke for obligatorisk felt skal vises */
  required?: boolean;
  /** aria-label for knapp i label */
  labelButtonAriaLabel?: string;
  /** Størrelse på feltnavn over radioknappene */
  labelSize?: 'small' | 'normal' | 'large';
  /** Overstyr label, se LabelWithCallout komponent */
  labelWithCalloutProps?: LabelWithCalloutProps;
  /** Brukerspesifisert event for callout **/
  onCalloutToggle?: (
    oldCalloutState: calloutState,
    newCalloutState: calloutState
  ) => void;
  options: IRadioButtonGroupOptions[];
  /** Callout warning */
  warning?: JSX.Element | string;
}
