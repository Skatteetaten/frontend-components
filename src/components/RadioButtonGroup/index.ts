export { default } from './RadioButtonGroup';
export * from './RadioButtonGroup';
