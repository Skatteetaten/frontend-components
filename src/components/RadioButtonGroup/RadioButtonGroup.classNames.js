import { mergeStyles } from '@uifabric/merge-styles';
import { getTheme } from '@uifabric/styling';
import { FontSizes, FontWeights } from '../utils/fonts';
import { isUndefined } from 'util';

export const getClassNames = props => {
  const theme = getTheme();
  const inset = -6;

  return mergeStyles({
    selectors: {
      '::-moz-focus-inner': {
        border: '0'
      },
      ' & .ms-ChoiceField-wrapper': {
        position: 'relative'
      },
      '& .ms-ChoiceField-field': {
        display: 'inline-block',
        marginTop: '8px',
        position: 'relative',
        outline: 0,
        verticalAlign: 'top'
      },
      'span.ms-ChoiceFieldLabel': {
        fontSize: FontSizes.medium,
        display: 'inline-block',
        paddingLeft: '26px'
      },
      '& .ms-ChoiceField-field::before': {
        content: '',
        display: 'inline-block',
        border: '2px solid' + theme.palette.skeColor.black,
        position: 'absolute'
      },
      '.ms-ChoiceField-input': {
        backgroundColor: theme.palette.skeColor.white,
        border: '2px solid' + theme.palette.skeColor.blue
      },
      '.ms-ChoiceField-input:disabled + .ms-ChoiceField-field::before': {
        color: theme.palette.skeColor.lightGrey,
        backgroundColor: theme.palette.skeColor.white,
        border: '2px solid' + theme.palette.skeColor.lightGrey
      },
      '.ms-ChoiceField-input:checked + .ms-ChoiceField-field::before': {
        backgroundColor: theme.palette.skeColor.white,
        border: '2px solid' + theme.palette.skeColor.blue
      },
      '.ms-Fabric.is-focusVisible.is-focusVisible & .ms-ChoiceField-wrapper.is-inFocus:after': {
        content: '""',
        position: 'absolute',
        left: inset,
        top: inset,
        bottom: inset,
        right: inset,
        outline: 'none',
        zIndex: 1
      },
      '& label.ms-Label': {
        fontSize: FontSizes.medium,
        fontWeight: FontWeights.regular
      },
      'span.ms-Label': {
        fontSize: FontSizes.medium,
        fontWeight: FontWeights.regular
      },
      '& .ms-Callout-main': !isUndefined(props.calloutFloating) &&
        !props.calloutFloating && {
          display: 'inline-block'
        }
    }
  });
};
