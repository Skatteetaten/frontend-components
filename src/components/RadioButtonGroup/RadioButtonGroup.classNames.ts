import { mergeStyles } from '@uifabric/merge-styles';
import { getTheme } from '@uifabric/styling';
import { FontSizes, FontWeights } from '../utils/fonts';
import { isUndefined } from 'util';
import { PaletteProps } from '..';
// @ts-ignore TODO
export const getClassNames = props => {
  const palette = getTheme().palette as PaletteProps;
  const inset = -6;

  return mergeStyles({
    selectors: {
      '::-moz-focus-inner': {
        border: '0'
      },
      ' & .ms-ChoiceField-wrapper': {
        position: 'relative'
      },
      '& .ms-ChoiceField-field': {
        display: 'inline-block',
        marginTop: '8px',
        position: 'relative',
        outline: 0,
        verticalAlign: 'top'
      },
      'span.ms-ChoiceFieldLabel': {
        fontSize: FontSizes.medium,
        display: 'inline-block',
        paddingLeft: '26px'
      },
      '& .ms-ChoiceField-field::before': {
        content: '',
        display: 'inline-block',
        border: '2px solid' + palette.skeColor.black,
        position: 'absolute'
      },
      '.ms-ChoiceField-input': {
        backgroundColor: palette.skeColor.white,
        border: '2px solid' + palette.skeColor.blue
      },
      '.ms-ChoiceField-input:disabled + .ms-ChoiceField-field::before': {
        color: palette.skeColor.lightGrey,
        backgroundColor: palette.skeColor.white,
        border: '2px solid' + palette.skeColor.lightGrey
      },
      '.ms-ChoiceField-input:checked + .ms-ChoiceField-field::before': {
        backgroundColor: palette.skeColor.white,
        border: '2px solid' + palette.skeColor.blue
      },
      '.ms-Fabric.is-focusVisible.is-focusVisible & .ms-ChoiceField-wrapper.is-inFocus:after': {
        content: '""',
        position: 'absolute',
        left: inset,
        top: inset,
        bottom: inset,
        right: inset,
        outline: 'none',
        zIndex: 1
      },
      '& label.ms-Label': {
        fontSize: FontSizes.medium,
        fontWeight: FontWeights.regular
      },
      'span.ms-Label': {
        fontSize: FontSizes.medium,
        fontWeight: FontWeights.regular
      },
      '& .ms-Callout-main': !isUndefined(props.calloutFloating) &&
        !props.calloutFloating && {
          display: 'inline-block'
        }
    }
  });
};
