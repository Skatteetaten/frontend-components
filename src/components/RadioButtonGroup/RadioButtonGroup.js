import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { ChoiceGroup as FabricChoiceGroup } from 'office-ui-fabric-react/lib-commonjs/ChoiceGroup';
import { getClassNames } from './RadioButtonGroup.classNames';
import ErrorMessage from '../ErrorMessage/ErrorMessage';
import { IconButton, Label } from 'office-ui-fabric-react/lib-commonjs';
import { getClassNames as getLabelClassNames } from '../TextField/TextFieldLabel.classNames';
import Callout from '../Callout';
import { isUndefined } from 'util';

/**
 * @visibleName RadioButtonGroup (Radioknapper)
 */

const RadioButtonGroup = props => {
  const {
    calloutFloating,
    children,
    className,
    errorMessage,
    help,
    id,
    label,
    warning,
    ...rest
  } = props;
  const [activeCallout, setActiveCallout] = React.useState('');
  const [isCalloutVisible, setIsCalloutVisible] = React.useState(false);
  const [_iconButtonElement, set_iconButtonElement] = React.useState('');
  const styles = getLabelClassNames(props);
  const helpElement = <p>{help}</p>;
  let warningElement = <p>{warning}</p>;

  return (
    <>
      <div id={id} className={styles.labelArea}>
        <span className={styles.label}>
          {label ? <Label>{label}</Label> : null}
        </span>
        {help && (
          <span
            className={styles.labelIconArea}
            ref={helpButton => set_iconButtonElement(helpButton)}
          >
            <IconButton
              iconProps={{ iconName: 'HelpOutline' }}
              title="Hjelp"
              ariaLabel={'Hjelpetekst til feltet ' + label}
              onClick={() => {
                setIsCalloutVisible(!isCalloutVisible);
                setActiveCallout('helpCallout');
              }}
              className={styles.icon}
            />
          </span>
        )}
        {warning && (
          <span
            className={styles.labelIconArea}
            ref={warningButton => set_iconButtonElement(warningButton)}
          >
            <IconButton
              iconProps={{ iconName: 'WarningOutline' }}
              title="Varsel"
              ariaLabel={'Varseltekst til feltet ' + label}
              onClick={() => {
                setIsCalloutVisible(!isCalloutVisible);
                setActiveCallout('warningCallout');
              }}
              className={styles.icon}
            />
          </span>
        )}
        {isCalloutVisible && (
          <Callout
            directionalHint={
              !isUndefined(calloutFloating) && !calloutFloating
                ? Callout.POS_BOTTOM_LEFT
                : Callout.POS_TOP_LEFT
            }
            color={
              activeCallout === 'helpCallout' ? Callout.HELP : Callout.WARNING
            }
            ariaLabel={
              activeCallout === 'helpCallout' ? 'Hjelpetekst' : 'Varseltekst'
            }
            target={_iconButtonElement}
            onClose={() => setIsCalloutVisible(false)}
          >
            {activeCallout === 'helpCallout' ? helpElement : warningElement}
          </Callout>
        )}
      </div>
      <FabricChoiceGroup
        {...rest}
        className={classnames(getClassNames(props), className)}
        ariaLabelledBy={label}
      >
        {children}
      </FabricChoiceGroup>
      {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
    </>
  );
};
RadioButtonGroup.propTypes = {
  /** Bestemmer om hjelptekst/varseltekst skal legge seg mellom label og tekstfelt eller flytende over innhold */
  calloutFloating: PropTypes.bool,
  /** Klassenavn **/
  className: PropTypes.string,
  /** Feilmelding **/
  errorMessage: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  /** Tilhørende hjelpetekst */
  help: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  /** Global attributt som må være unik for hele HTML dokumentet */
  id: PropTypes.string,
  /** Label */
  label: PropTypes.string,
  /** Tilhørende varseltekst */
  warning: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
};

export default RadioButtonGroup;
