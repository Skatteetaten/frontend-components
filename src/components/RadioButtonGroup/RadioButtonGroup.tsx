import classnames from 'classnames';
import { IconButton, Label } from 'office-ui-fabric-react/lib-commonjs';
import {
  ChoiceGroup as FabricChoiceGroup,
  IChoiceGroupProps
} from 'office-ui-fabric-react/lib-commonjs/ChoiceGroup';
import * as React from 'react';
import { isUndefined } from 'util';
import Callout from '../Callout';
import ErrorMessage from '../ErrorMessage/ErrorMessage';
import { getClassNames as getLabelClassNames } from '../TextField/TextFieldLabel.classNames';
import { getClassNames } from './RadioButtonGroup.classNames';

interface RadioButtonGroupProps extends IChoiceGroupProps {
  calloutFloating?: boolean;
  className?: string;
  errorMessage?: JSX.Element | string;
  help?: JSX.Element | string;
  warning: JSX.Element | string;
}

/**
 * @visibleName RadioButtonGroup (Radioknapper)
 */

const RadioButtonGroup = (props: RadioButtonGroupProps) => {
  const {
    calloutFloating,
    children,
    className,
    errorMessage,
    help,
    id,
    label,
    warning,
    ...rest
  } = props;
  const [activeCallout, setActiveCallout] = React.useState('');
  const [isCalloutVisible, setIsCalloutVisible] = React.useState(false);
  const [_iconButtonElement, set_iconButtonElement] = React.useState('');
  const styles = getLabelClassNames(props);
  const helpElement = <p>{help}</p>;
  let warningElement = <p>{warning}</p>;

  return (
    <>
      <div id={id} className={styles.labelArea}>
        <span className={styles.label}>
          {label ? <Label>{label}</Label> : null}
        </span>
        {help && (
          <span
            className={styles.labelIconArea}
            // @ts-ignore TODO
            ref={helpButton => set_iconButtonElement(helpButton)}
          >
            <IconButton
              iconProps={{ iconName: 'HelpOutline' }}
              title="Hjelp"
              ariaLabel={'Hjelpetekst til feltet ' + label}
              onClick={() => {
                setIsCalloutVisible(!isCalloutVisible);
                setActiveCallout('helpCallout');
              }}
              className={styles.icon}
            />
          </span>
        )}
        {warning && (
          <span
            className={styles.labelIconArea}
            // @ts-ignore TODO
            ref={warningButton => set_iconButtonElement(warningButton)}
          >
            <IconButton
              iconProps={{ iconName: 'WarningOutline' }}
              title="Varsel"
              ariaLabel={'Varseltekst til feltet ' + label}
              onClick={() => {
                setIsCalloutVisible(!isCalloutVisible);
                setActiveCallout('warningCallout');
              }}
              className={styles.icon}
            />
          </span>
        )}
        {isCalloutVisible && (
          <Callout
            directionalHint={
              !isUndefined(calloutFloating) && !calloutFloating
                ? Callout.POS_BOTTOM_LEFT
                : Callout.POS_TOP_LEFT
            }
            color={
              activeCallout === 'helpCallout' ? Callout.HELP : Callout.WARNING
            }
            ariaLabel={
              activeCallout === 'helpCallout' ? 'Hjelpetekst' : 'Varseltekst'
            }
            target={_iconButtonElement}
            onClose={() => setIsCalloutVisible(false)}
          >
            {activeCallout === 'helpCallout' ? helpElement : warningElement}
          </Callout>
        )}
      </div>
      <FabricChoiceGroup
        {...rest}
        className={classnames(getClassNames(props), className)}
        ariaLabelledBy={label}
      >
        {children}
      </FabricChoiceGroup>
      {/* 
      // @ts-ignore TODO */}
      <ErrorMessage showError={!!errorMessage}>{errorMessage}</ErrorMessage>
    </>
  );
};

export default RadioButtonGroup;
