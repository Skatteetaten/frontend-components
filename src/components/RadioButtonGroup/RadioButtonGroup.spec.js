import React from 'react';
import { mount } from 'enzyme';
import { matches } from './../utils/test-utils';
import RadioButtonGroup from './RadioButtonGroup';

const options = [
  {
    key: 'A',
    text: 'Valg A'
  },
  {
    key: 'B',
    text: 'Valg B'
  },
  {
    key: 'C',
    text: 'Valg C',
    disabled: true
  },
  {
    key: 'D',
    text: 'Valg D',
    disabled: true
  }
];

function oppsettFullDOM(props) {
  return mount(<RadioButtonGroup {...props} />);
}

describe('RadioButtonGroup komponent', () => {
  it('matcher snapshot', () => {
    matches(
      <RadioButtonGroup
        required={true}
        label={'Velg en'}
        defaultSelectedKey={'B'}
        options={options}
      />
    );
  });

  it('rendrer RadioButtonGroup med riktig props', () => {
    const wrapper = oppsettFullDOM({
      required: true,
      id: 'radiobuttongroup-id',
      className: 'radiobuttongroup-class',
      label: 'Velg en',
      defaultSelectedKey: 'B',
      options: options
    });
    expect(
      wrapper
        .find('div')
        .first()
        .prop('id')
    ).toEqual('radiobuttongroup-id');
    expect(wrapper.prop('className')).toEqual('radiobuttongroup-class');
    expect(
      wrapper.find('StyledChoiceGroupBase').props().ariaLabelledBy
    ).toEqual('Velg en');
    expect(wrapper.find('StyledChoiceGroupBase').prop('required')).toEqual(
      true
    );
    expect(wrapper.exists('#ChoiceGroup3-B')).toEqual(true);
    expect(wrapper.find('input#ChoiceGroup3-B').prop('checked')).toEqual(true);
    expect(wrapper.find('input#ChoiceGroup3-B').prop('disabled')).toEqual(
      undefined
    );
    expect(wrapper.find('input#ChoiceGroup3-C').prop('checked')).toEqual(false);
    expect(wrapper.find('input#ChoiceGroup3-C').prop('disabled')).toEqual(true);
  });

  it('rendrer RadioButtonGroup med riktig antall valg og innhold', () => {
    const wrapper = oppsettFullDOM({
      required: true,
      id: 'radiobuttongroup-id',
      className: 'radiobuttongroup-class',
      label: 'Velg en',
      defaultSelectedKey: 'B',
      options: options
    });

    const radiobutton = wrapper.find('StyledChoiceGroupOptionBase');
    expect(radiobutton).toHaveLength(4);
    expect(
      radiobutton
        .at(2)
        .find('.ms-ChoiceFieldLabel')
        .html()
    ).toContain('Valg C');
  });
});
