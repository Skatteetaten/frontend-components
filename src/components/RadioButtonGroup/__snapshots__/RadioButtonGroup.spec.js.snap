// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup komponent matcher snapshot 1`] = `
<fieldset
  className="fieldset-42"
>
  <legend
    className="labelAsLegend-46"
  >
    Velg en
  </legend>
  <div
    className="radioButtons-43"
  >
    <div
      aria-labelledby="Velg en"
      className="ms-ChoiceFieldGroup root-51"
      role="radiogroup"
    >
      <div
        className="ms-ChoiceFieldGroup-flexContainer"
      >
        <div
          className="ms-ChoiceField root-52"
        >
          <div
            className="ms-ChoiceField-wrapper"
          >
            <input
              checked={false}
              className="ms-ChoiceField-input input-53"
              id="ChoiceGroup1-A"
              name="ChoiceGroup1"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="radio"
            />
            <label
              className="ms-ChoiceField-field field-54"
              htmlFor="ChoiceGroup1-A"
            >
              <span
                className="ms-ChoiceFieldLabel"
                id="ChoiceGroupLabel2-A"
              >
                Valg A
              </span>
            </label>
          </div>
        </div>
        <div
          className="ms-ChoiceField root-52"
        >
          <div
            className="ms-ChoiceField-wrapper"
          >
            <input
              checked={true}
              className="ms-ChoiceField-input input-53"
              id="ChoiceGroup1-B"
              name="ChoiceGroup1"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="radio"
            />
            <label
              className="ms-ChoiceField-field is-checked field-59"
              htmlFor="ChoiceGroup1-B"
            >
              <span
                className="ms-ChoiceFieldLabel"
                id="ChoiceGroupLabel2-B"
              >
                Valg B
              </span>
            </label>
          </div>
        </div>
        <div
          className="ms-ChoiceField root-52"
        >
          <div
            className="ms-ChoiceField-wrapper"
          >
            <input
              checked={false}
              className="ms-ChoiceField-input is-disabled input-53"
              disabled={true}
              id="ChoiceGroup1-C"
              name="ChoiceGroup1"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="radio"
            />
            <label
              className="ms-ChoiceField-field field-62"
              htmlFor="ChoiceGroup1-C"
            >
              <span
                className="ms-ChoiceFieldLabel"
                id="ChoiceGroupLabel2-C"
              >
                Valg C
              </span>
            </label>
          </div>
        </div>
        <div
          className="ms-ChoiceField root-52"
        >
          <div
            className="ms-ChoiceField-wrapper"
          >
            <input
              checked={false}
              className="ms-ChoiceField-input is-disabled input-53"
              disabled={true}
              id="ChoiceGroup1-D"
              name="ChoiceGroup1"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="radio"
            />
            <label
              className="ms-ChoiceField-field field-62"
              htmlFor="ChoiceGroup1-D"
            >
              <span
                className="ms-ChoiceFieldLabel"
                id="ChoiceGroupLabel2-D"
              >
                Valg D
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</fieldset>
`;
